### PR TITLE
Add /server-info endpoint

### LINKED ISSUE #20: add /server-info end point providing machine-readable information
ATM in `dandi-cli` I need to
- [hardcode](https://github.com/dandi/dandi-cli/blob/master/dandi/consts.py#L71) mapping for deployments (local, main) for corresponding addresses of underlying girder service
- embed similar mapping for netlify instances which uses main girder instance 
- support a cacophony of url mappings for downloader support (this hopefully would be gone when we get our API going)

I think it would be nice if server could be queried relevant information. I see following (assuming json, with non-json comments ;-)) structure could be returned now
```json
{
  "version": 1,  # version of this spec
  "api-versions": [],  # Later 1, 2 and much later 2, 3, 4
  "cli-minimal-version": "0.6.0",
  "cli-bad-versions": [],  # 
  "services": {
      "girder": {
          "url": "https://girder.dandiarchive.org"
       },
       "webui": {
          "url": "https://gui.dandiarchive.org"
       },
       "api": {
           "url": "https://publish.dandiarchive.org/api"
       },
       "jupyterhub": {
          "url": "https://hub.dandiarchive.org"
       }
}
```
Then any specific deployment could provide its own details, so clients could avoid any hardcoding, work with any deployment, and provide user with an informative error if they encounter some version they do not support, or a deployment lacking a specific feature/service.

May be there is a better, standardized schema for such descriptions already which we could adopt.

### PR DESCRIPTION
**Pull request by jwodder**
Closes #20

### CODE REVIEW

**Comment by yarikoptic** [comment]
Looks good to me, thank you @jwodder ! I will leave it to @satra for the final verdict.  E.g. here may be a pure `/info` might be more adequate (`/server-info` I dragged from the original `dandiarchive` as analog to apache's one)

File: serve.py [hunk 1]
```diff
@@ -7,9 +7,15 @@
...
 GIRDER_URL = os.environ.get("GIRDER_URL", "https://girder.dandiarchive.org").rstrip('/')
+
 GUI_URL = os.environ.get("GUI_URL", "https://gui.dandiarchive.org").rstrip('/')
+
 ABOUT_URL = os.environ.get("ABOUT_URL", "https://www.dandiarchive.org").rstrip('/')
 
+PUBLISH_API_URL = os.environ.get("PUBLISH_API_URL", "https://publish.dandiarchive.org/api").rstrip()
+
+JUPYTERHUB_URL = os.environ.get("JUPYTERHUB_URL", "https://hub.dandiarchive.org").rstrip()
+
```
Comment (satra) [inline serve.py:18]: perhaps we just go with pre-commit (with black) across all dandi codebases. @yarikoptic @mgrauer ??

**Comment by satra** [comment]
and deployed!

File: serve.py (hunk 1 above)
Comment (yarikoptic) [reply serve.py:18]: I don't mind - better be done now while it is short code base, or not actively worked on with outstanding PRs ;-)

**Comment by yarikoptic** [comment]
Awesome, thank you @satra and @jwodder !

File: serve.py [hunk 2]
```diff
@@ -136,6 +142,29 @@ async def goto_dandiset_version(request, dataset, version):
...
+                "url": GUI_URL,
+            },
+            "api": {
+                "url": PUBLISH_API_URL,
+            },
+            "jupyterhub": {
+                "url": JUPYTERHUB_URL,
+            }
+        }
+    })
```
Comment (yarikoptic) [inline serve.py:166]: Minor: now seeing it in action realized that it should've been indented - there is no traffic to spare

Also I wonder - why it escaped all `/` - is it needed?

```
{"version":"1.0.0","cli-minimal-version":"0.5.0","cli-bad-versions":[],"services":{"girder":{"url":"https:\/\/girder.dandiarchive.org"},"webui":{"url":"https:\/\/gui.dandiarchive.org"},"api":{"url":"https:\/\/publish.dandiarchive.org\/api"},"jupyterhub":{"url":"https:\/\/hub.dandiarchive.org"}}}
```

Comment (jwodder) [reply serve.py:166]: JSON allows escaping of `/`, apparently in order to avoid cross-site scripting attacks somehow.

**Pull request merged**